dnfdaemon-server/services/rpm/rpm.cpp

```diff
@@ -123,54 +130,60 @@ void Rpm::dbus_register() {
 }
 
 void Rpm::list(sdbus::MethodCall && call) {
-    auto worker = std::thread([this](sdbus::MethodCall call) {
-        try {
-            // read options from dbus call
-            dnfdaemon::KeyValueMap options;
-            call >> options;
-
-            session.fill_sack();
-            auto & solv_sack = session.get_base()->get_rpm_solv_sack();
-
-            // patterns to search
-            std::vector<std::string> default_patterns{};
-            std::vector<std::string> patterns =
-                key_value_map_get<std::vector<std::string>>(options, "patterns", std::move(default_patterns));
-
-            libdnf::rpm::PackageSet result_pset(&solv_sack);
-            libdnf::rpm::SolvQuery full_solv_query(&solv_sack);
-            if (patterns.size() > 0) {
-                for (auto & pattern : patterns) {
-                    libdnf::rpm::SolvQuery solv_query(full_solv_query);
-                    solv_query.resolve_pkg_spec(pattern, true, true, true, true, true, {});
-                    result_pset |= solv_query;
+    auto worker = std::thread(
+        [this](sdbus::MethodCall call) {
+            try {
+                // read options from dbus call
+                dnfdaemon::KeyValueMap options;
+                call >> options;
+
+                session.fill_sack();
+                auto & solv_sack = session.get_base()->get_rpm_solv_sack();
+
+                // patterns to search
+                std::vector<std::string> default_patterns{};
+                std::vector<std::string> patterns =
+                    key_value_map_get<std::vector<std::string>>(options, "patterns", std::move(default_patterns));
+
+                libdnf::rpm::PackageSet result_pset(&solv_sack);
+                libdnf::rpm::SolvQuery full_solv_query(&solv_sack);
+                if (patterns.size() > 0) {
+                    for (auto & pattern : patterns) {
+                        libdnf::rpm::SolvQuery solv_query(full_solv_query);
+                        solv_query.resolve_pkg_spec(pattern, true, true, true, true, true, {});
```

Comment:
It looks like that `dnf list <provide>` provides no results but your code will find it. I am not sure whether it is a required difference or not. What do you think?